src/main/java/ch/njol/skript/util/chat/ChatMessages.java

```diff
@@ -579,19 +578,23 @@ public static void registerAddonCode(@Nullable SkriptAddon addon, @Nullable Chat
 	 * @return A string without styles.
 	 */
 	public static String stripStyles(String text) {
-		List<MessageComponent> components = parse(text);
-		StringBuilder sb = new StringBuilder();
-		for (MessageComponent component : components) { // This also strips bracket tags ex. <red> <ttp:..> etc.
-			sb.append(component.text);
-		}
-		String plain = sb.toString();
-
-		if (Utils.HEX_SUPPORTED) // Strip '§x', '&x'
-			plain = plain.replaceAll("[§&]x", "");
-
-		plain = plain.replaceAll("(?i)[&§][0-9a-folkrnm]", ""); // strips colors & or § (ex. &5)
-
-		assert plain != null;
-		return plain;
+		String previous;
+		String result = text;
+		do {
+			previous = result;
+			
+			List<MessageComponent> components = parse(result);
+			StringBuilder builder = new StringBuilder();
```

Comment:
Isn't it better to initialize the builder outside the loop and reset it inside the loop every iteration?

Reply:
Probably no performance difference (JIT compiler), and if there was it would be ~10ns.